### PR TITLE
[mimic] Sync up ceph-mgr prometheus related changes

### PR DESCRIPTION
**Pull request by b-ranto**
This syncs the mimic branch with the master branch with regard to the ceph-mgr prometheus module. All the patches apply cleanly since the branches didn't diverge much, yet.

I have made a first rados test run for this without the patch that fixes a bug in ceph-mgr dashboard, here:
http://pulpito.ceph.com/branto-2018-05-30_20:06:42-rados-wip-branto-mgr-mimic-distro-basic-smithi/

Since I noticed a couple of failures that this exposed in the dashboard module, I have also back-ported the dashboard fix and re-run rados:mgr test suite:
http://pulpito.ceph.com/branto-2018-05-31_09:48:16-rados:mgr-wip-branto-mgr-mimic-distro-basic-smithi/

### CODE REVIEW

**Comment by b-ranto** [comment]
This will also require the PR #22420 that fixes a bug in exposing the data to the mgr modules. Marking DNM in the meantime.

**Comment by b-ranto** [comment]
The PR was merged and back-ported, removing DNM.

**Comment by b-ranto** [comment]
retest this please

File: src/pybind/mgr/prometheus/module.py [hunk 1]
```diff
@@ -552,25 +567,47 @@ def collect(self):
...
+                # Get the value of the counter
+                value = self._perfvalue_to_value(counter_info['type'], counter_info['value'])
+
+                # Represent the long running avgs as sum/count pairs
+                if counter_info['type'] & self.PERFCOUNTER_LONGRUNAVG:
+                    _path = path + '_sum'
+                    self.metrics.add_metric(_path, Metric(
+                        stattype,
+                        _path,
+                        counter_info['description'] + ' Total',
```
Comment (votdev) [inline src/pybind/mgr/prometheus/module.py]: You're using ``"xxxx {} yyy".format(zzz)``, why not here?

File: src/pybind/mgr/prometheus/module.py [hunk 2]
```diff
@@ -552,25 +567,47 @@ def collect(self):
...
                     self.log.debug('ignoring %s, type %s' % (path, stattype))
                     continue
 
-                self.metrics.add_metric(path, Metric(
+                # Get the value of the counter
+                value = self._perfvalue_to_value(counter_info['type'], counter_info['value'])
+
+                # Represent the long running avgs as sum/count pairs
+                if counter_info['type'] & self.PERFCOUNTER_LONGRUNAVG:
+                    _path = path + '_sum'
```
Comment (votdev) [inline src/pybind/mgr/prometheus/module.py]: You're using ``"xxxx {} yyy".format(zzz)``, why not here?

File: src/pybind/mgr/prometheus/module.py [hunk 3]
```diff
@@ -552,25 +567,47 @@ def collect(self):
...
+                        counter_info['description'] + ' Total',
+                        ("ceph_daemon",),
+                    ))
+                    self.metrics.append(_path, value, (daemon,))
+
+                    _path = path + '_count'
+                    self.metrics.add_metric(_path, Metric(
+                        'counter',
+                        _path,
+                        counter_info['description'] + ' Count',
```
Comment (votdev) [inline src/pybind/mgr/prometheus/module.py]: You're using ``"xxxx {} yyy".format(zzz)``, why not here?

File: src/pybind/mgr/prometheus/module.py [hunk 4]
```diff
@@ -552,25 +567,47 @@ def collect(self):
...
+                    _path = path + '_sum'
+                    self.metrics.add_metric(_path, Metric(
+                        stattype,
+                        _path,
+                        counter_info['description'] + ' Total',
+                        ("ceph_daemon",),
+                    ))
+                    self.metrics.append(_path, value, (daemon,))
+
+                    _path = path + '_count'
```
Comment (votdev) [inline src/pybind/mgr/prometheus/module.py]: You're using ``"xxxx {} yyy".format(zzz)``, why not here?

**Comment by b-ranto** [comment]
@votdev This is a back-port, I would prefer not to diverge from master in the back-port.

Anyway, I usually use '+' for concatenation of two strings and '%' , 'format' or 'join' for multiple strings. I find it easier to read it that way for two strings and there is pretty much no performance penalty. I may have used 'format' in one of the commits because it did fit the style in that place more though (i.e. because other people used it in the surrounding code).

**Pull request merged**